packages/data-service/src/data-service.ts

```diff
@@ -133,6 +134,26 @@ function isReadPreferenceSet(connectionString: string): boolean {
   );
 }
 
+function readPreferenceWithoutTags(readPreference: ReadPreference) {
+  return new ReadPreference(readPreference.mode, undefined, {
```

Comment:
I can't spot a way to be future-proof here. ie. make a read preference that is exactly the same but just with the tags removed.